File: Java/9.回文数.java
/*
 * @lc app=leetcode.cn id=9 lang=java
 *
 * [9] 回文数
 */

// @lc code=start
class Solution {
    public boolean isPalindrome(int x) {
        Boolean res = true;

        String str = String.valueOf(x);

        for (int i = 0; i < str.length() / 2 && x >= 0; i++) {
            if (str.charAt(i) != str.charAt(str.length() - i -1))
                res = false;
        }
        
        if (x < 0)
            res = false;

        return res;
    }
}
// @lc code=end

File: Python/9.回文数.py
#
# @lc app=leetcode.cn id=9 lang=python3
#
# [9] 回文数
#

# @lc code=start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        res = True
        string = str(x)
        length = len(string)

        if x >= 0:
            for i in range(length // 2):
                if string[i] != string[length - i - 1]:
                    res = False
        else:
            res = False

        return res   
# @lc code=end

